lib/commands/debug/helper.js

```diff
@@ -0,0 +1,143 @@
+    });
+}
+
+function invokePythonDebugger(token, localDebuggerPath, skillId, handlerName, skillFilePath) {
+    nodemon({
+        execMap: {
+            py: 'python3'
+        },
+        script: localDebuggerPath,
+        args: ['--accessToken', `"${token}"`, '--skillId', skillId,
+            '--skillHandler', handlerName, '--skillFilePath', skillFilePath],
+        ext: 'py,json,txt'
+    });
+}
+
+function invokeAttachPythonDebugger(debugPort, token, localDebuggerPath, skillId, handlerName, skillFilePath) {
+    execSync('python3 -m pip install debugpy', { stdio: 'inherit' });
+    nodemon({
+        execMap: {
+            py: `python3 -m debugpy --listen ${debugPort} --wait-for-client`
+        },
+        script: localDebuggerPath,
+        args: ['--accessToken', `"${token}"`, '--skillId', skillId,
+            '--skillHandler', handlerName, '--skillFilePath', skillFilePath],
+        ext: 'py,json,txt'
+    });
+}
+
+function getSkillInvocationInfo(handler, runtime, isHosted) {
+    if (runtime.includes('node') || runtime.includes('python')) {
+        if (isHosted) {
+            if (runtime.includes('node')) {
+                return { handlerName: 'handler', skillFileName: 'index' };
+            }
+            if (runtime.includes('python')) {
+                return { handlerName: 'lambda_handler', skillFileName: 'lambda_function' };
+            }
+        }
+        const handlerInfo = handler.split('.');
+        const handlerName = handlerInfo.pop();
+        const skillFileName = handlerInfo.join(path.delimiter);
+        return { handlerName, skillFileName };
+    }
+    if (runtime.includes('java')) {
+        return { handlerName: handler };
+    }
+}
+
+function initiateDebugging(runtime, skillInvocationInfo, skillCodeFolderName, waitForAttach, debugPort, token, skillId) {
+    if (runtime.includes('python')) {
+        const localDebuggerPath = getPythonDebugAdapterPath();
+        if (localDebuggerPath === undefined) {
+            throw new CliError('Install ask-sdk-local-debug');
+        }
+        if (waitForAttach) {
+            invokeAttachPythonDebugger(debugPort, token, localDebuggerPath, skillId, skillInvocationInfo.handlerName,
+                `${skillCodeFolderName}/${skillInvocationInfo.skillFileName}.py`);
+        } else {
+            invokePythonDebugger(token, localDebuggerPath, skillId, skillInvocationInfo.handlerName,
+                `${skillCodeFolderName}/${skillInvocationInfo.skillFileName}.py`);
+        }
+    } else if (runtime.includes('node')) {
+        if (waitForAttach) {
+            invokeAttachNodeDebugger(debugPort, token, skillId, skillCodeFolderName,
+                `${skillCodeFolderName}/${skillInvocationInfo.skillFileName}.js`, skillInvocationInfo.handlerName);
+        } else {
+            invokeNodeDebugger(token, skillId, skillCodeFolderName,
+                `${skillCodeFolderName}/${skillInvocationInfo.skillFileName}.js`, skillInvocationInfo.handlerName);
```

Comment:
one overall comment regarding the language debugger. I see invoke{Lang} and invokeAttach{Lang} as two modes which are sharing the same input to the function. Do you think this indicates we need to create a CliDebugger class so that different languages are based on same interfaces? invoke or invokeAttach share the same interfaces.

If we have that, here we only need to write:
```
new CliDebugger(runtime);
if (wait) { 
  CliDebugger.invokeProcess();
} else {
  CliDebugger.invokeAttach();
}
```